src/sssom/util.py

```diff
@@ -393,6 +393,56 @@ def condense(self) -> List[str]:
         self.df.drop(columns=condensed, inplace=True)
         return condensed
 
+    def get_compatible_version(self) -> str:
+        """Get the minimum version of SSSOM this set is compatible with."""
+        schema = SSSOMSchemaView()
+        versions = set()
+
+        # First get the minimum versions required by the slots present
+        # in the set; this is entirely provided by the SSSOM model.
+        for slot in self.metadata.keys():
+            version = schema.get_minimum_version(slot, "mapping set")
+            if version is not None:
+                versions.add(str(version))
+        for slot in self.df.columns:
+            version = schema.get_minimum_version(slot, "mapping")
+            if version is not None:
+                versions.add(str(version))
+
+        # Then take care of enum values; we cannot use the SSSOM model
+        # for that (enum values are not tagged with an "added_in"
+        # annotation the way slots are), so this has to be handled
+        # "manually" based on the informations provided in
+        # <https://mapping-commons.github.io/sssom/spec-model/#model-changes-across-versions>.
+        if (
+            self.metadata.get("subject_type") == "composed entity expression"
+            or self.metadata.get("subject_type") == "composed entity expression"
+            or (
+                "subject_type" in self.df.columns
+                and "composed entity expression" in self.df["subject_type"].values
+            )
+            or (
+                "object_type" in self.df.columns
+                and "composed entity expression" in self.df["object_type"].values
+            )
+        ):
+            versions.add("1.1")
+
+        if (
+            "mapping_cardinality" in self.df.columns
+            and "0:0" in self.df["mapping_cardinality"].values
+        ):
+            versions.add("1.1")
+
+        # Get the highest of the accumulated versions. We do a numerical
+        # sort, so that version 1.10 (if we ever get that far in the 1.x
+        # branch) does not get sorted before version 1.9.
+        def _version_to_compare_key(version):
+            major, minor = [int(s) for s in version.split(".")]
+            return (major * 100) + minor
```

Comment:
Python is pretty happy to compare tuples in the way you'd expect them to, so this can be simplified to:

```suggestion
            return tuple(int(s) for s in version.split("."))
```

Reply:
Nice, thanks for the tip!